kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/kms/aws/AbstractAwsKmsTestKekManager.java

```diff
@@ -21,6 +21,7 @@
 
 public abstract class AbstractAwsKmsTestKekManager implements TestKekManager {
     private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private DescribeKeyResponse keyResponse;
```

Comment:
This is not a good approach.  You've are assuming an **order** to which the client calls the API methods.  Invocation order is not part of the contract. The caller is entitled to call the API methods however they please. 

Think about the cases it breaks:

* use cases involve >1 KEKs.
* use cases where the caller know _a priori_ that the alias exists so doesn't bother to call `exist()`.
* use cases where the they've somehow scaled their deployment.. one replica receives the `#exist()` call and another replica receives the `#rotate()` call.

To be honest, I don't really see a problem to solve.  KEK operations are low volume and low frequency. A few extra calls to read are tolerable. 

_If_ we did really care (if it were a hot path), you could maintain a thread safe cache (caffeine would work alias -> id) which you could use to short circuit the lookup.  But like I say, I really don't see the need.

Reply:
ok, thanks!